synapse/federation/transport/server/_base.py

```diff
@@ -373,6 +373,15 @@ def register(self, server: HttpServer) -> None:
             if code is None:
                 continue
 
+            if is_method_cancellable(code):
+                # The wrapper added by `self._wrap` will inherit the cancellable flag,
+                # but the wrapper itself does not support cancellation yet.
+                raise Exception(
```

Comment:
On one of the other PRs, I wondered (but didn't say) if we could use a more specific exception type here. I don't think there's an obvious candidate though, and if this gets thrown it'll happen loudly at startup, so it's probably fine as-is.

Reply:
I'm not sure which exception type would be more appropriate. Nothing really springs to mind.